src/pip/_internal/commands/install.py

```diff
@@ -321,21 +315,51 @@ def run(self, options, args):
                         modifying_pip=requirement_set.has_requirement("pip")
                     )
 
-                    # If caching is disabled or wheel is not installed don't
-                    # try to build wheels.
-                    if wheel and options.cache_dir:
-                        # build wheels before install.
-                        wb = WheelBuilder(
-                            finder, preparer, wheel_cache,
-                            build_options=[], global_options=[],
-                        )
-                        # Ignore the result: a failed wheel will be
-                        # installed from the sdist/vcs whatever.
+                    # Consider legacy and PEP517-using requirements separately
```

Comment:
Considering how long this has become, could this build-the-wheels logic be moved into a method on the command?

Reply:
That's a good idea. I'm focusing on getting the tests to pass right now, but refactoring things is something that definitely needs to be done.

Reply:
Now would be a nice time to approach refactoring this. 

I'm okay if you want to tackle the refactoring in a follow up; tbh I won't mind doing it myself.